.github/workflows/ceph-s3-tests.yaml

```diff
@@ -0,0 +1,78 @@
+name: ceph-s3-tests
+on: 
+  push:
+    branches:
+      - master
+    paths:
+      - 'src/*'
+  pull_request:
+      branches:
+      - master
+      paths:
+      - 'src/*'
+  workflow_dispatch:
+
+
+jobs:
+  ceph-s3-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout noobaa-core
+        uses: actions/checkout@v3
+        with:
+          repository: 'noobaa/noobaa-core'
+          path: 'noobaa-core'
+
+      - name: Deploy minikube
+        run: |
+         cd ./noobaa-core
+         sudo bash ./.travis/deploy_minikube.sh
+
+      - name: Build noobaa tester
+        run: |
+         cd ./noobaa-core
+         make tester TESTER_TAG=noobaa-tester:s3-tests
+         docker tag noobaa:latest noobaa-core:s3-tests
+
+      - name: Checkout noobaa-operator
+        uses: actions/checkout@v3
+        with:
+          repository: 'noobaa/noobaa-operator'
+          path: 'noobaa-operator'
+
+      - name: Update kubectl context
+        run: |
+          sudo mv /root/.kube /root/.minikube $HOME
+          sudo chown -R $USER $HOME/.kube $HOME/.minikube
+          sed "s/root/home\/$USER/g" $HOME/.kube/config > tmp; mv tmp $HOME/.kube/config
+
+      - name: Build operator
+        run: |
+         cd ./noobaa-operator
+         make all
+
+      - name: Install noobaa system
+        run: |
+         cd ./noobaa-operator
+         ./build/_output/bin/noobaa-operator install --mini --noobaa-image='noobaa-core:s3-tests'
+
+      - name: Run Ceph s3-tests
+        run: |
+          set -x
+          sleep 3m
```

Comment:
Let's replace it with waiting for the noobaa-default-backing-store condition=available
```suggestion
          kubectl wait --for=condition=available backingstore/noobaa-default-backing-store --timeout=3m
```

Reply:
Hi,
It is not sufficient to wait for that condition (All the tests failed on my local machine).
The printings when all the tests are failing are:
`{ name: 'noobaa-default-backing-store', resource_type: 'HOSTS', pool_node_type: 'BLOCK_STORE_FS', storage: { total: 84223205376, free: 0, used: 0, reserved: 104857600, unavailable_free: 19406385152, unavailable_used: 0, used_other: 64711962624 }, create_time: 1672644959000, io_stats: { read_count: 0, write_count: 0, read_bytes: 0, write_bytes: 0 }, data_activities: { activities: [], host_count: 0 }, nodes: { count: 1, online: 1, by_mode: { INITIALIZING: 1 } }, storage_nodes: { count: 1, online: 1, by_mode: { INITIALIZING: 1 } }, s3_nodes: { count: 0, online: 0, by_mode: {} }, hosts: { count: 1, by_mode: { INITIALIZING: 1 }, by_service: { STORAGE: 1 }, configured_count: 1 }, host_info: { volume_size: 53687091200 }, is_managed: true, undeletable: 'IN_USE', mode: 'INITIALIZING', associated_accounts: [] }`
The printings when it's works fine:
`{ name: 'noobaa-default-backing-store', resource_type: 'HOSTS', pool_node_type: 'BLOCK_STORE_FS', storage: { total: 84223205376, free: 24484839424, used: 0, reserved: 104857600, unavailable_free: 0, unavailable_used: 0, used_other: 59633508352 }, create_time: 1672241677000, io_stats: { read_count: 10, write_count: 10, read_bytes: 10240, write_bytes: 10240 }, data_activities: { activities: [], host_count: 0 }, nodes: { count: 1, online: 1, by_mode: { OPTIMAL: 1 } }, storage_nodes: { count: 1, online: 1, by_mode: { OPTIMAL: 1 } }, s3_nodes: { count: 0, online: 0, by_mode: {} }, hosts: { count: 1, by_mode: { OPTIMAL: 1 }, by_service: { STORAGE: 1 }, configured_count: 1 }, host_info: { volume_size: 53687091200 }, is_managed: true, undeletable: 'IN_USE', mode: 'LOW_CAPACITY', associated_accounts: [] }`

We can see that the free storage is 0, and the mode is INITIALIZING.

Reply:
I also tried to wait for the moment the backing store is ready (manually) and then run all the tests, but still, all the tests were failing.

From nb system:
nb backingstore list
NAME                           TYPE      TARGET-BUCKET   PHASE   AGE
noobaa-default-backing-store   pv-pool                   Ready   1m26s

The printings from the job logs:
 `{ name: 'noobaa-default-backing-store', resource_type: 'HOSTS', pool_node_type: 'BLOCK_STORE_FS', storage: { total: 84223205376, free: 0, used: 0, reserved: 104857600, unavailable_free: 16048295936, unavailable_used: 0, used_other: 68070051840 }, create_time: 1672658604000, io_stats: { read_count: 0, write_count: 0, read_bytes: 0, write_bytes: 0 }, data_activities: { activities: [], host_count: 0 }, nodes: { count: 1, online: 1, by_mode: { INITIALIZING: 1 } }, storage_nodes: { count: 1, online: 1, by_mode: { INITIALIZING: 1 } }, s3_nodes: { count: 0, online: 0, by_mode: {} }, hosts: { count: 1, by_mode: { INITIALIZING: 1 }, by_service: { STORAGE: 1 }, configured_count: 1 }, host_info: { volume_size: 53687091200 }, is_managed: true, undeletable: 'IN_USE', mode: 'INITIALIZING', associated_accounts: [] }
`

We can see that the free storage is 0, and the mode is INITIALIZING.
and this time I think it's an issue... (phase is ready but it's not actually is...).
@dannyzaken, @liranmauda WDYT?

Reply:
I will share another trial I had with the code from this branch:
[The first](https://github.com/shirady/learn-action/actions/runs/3836247315/jobs/6530210631) - using sleep after waiting for the backing store to be ready. here is the printing of test pool:
![Screen Shot 2023-01-04 at 10 57 30](https://user-images.githubusercontent.com/57721533/210519173-5e455e99-c692-4b8e-b6ed-4ac504a60756.png)
[The second](https://github.com/shirady/learn-action/actions/runs/3836403533/jobs/6530533780) - without sleep, all the tests are failing!
![Screen Shot 2023-01-04 at 10 58 37](https://user-images.githubusercontent.com/57721533/210519408-a0a51533-d4af-4345-bd00-4c972fa5f0e8.png)
Note: The links are to a temp repo and will not be available after this PR will be closed.

Reply:
@Liran, your suggestion is great, but there's an [issue](https://github.com/noobaa/noobaa-operator/issues/1007) we need to fix to use it.
Meanwhile, I left the sleep command in the action with a comment beside it.